frontend/src/features/leases/detail/LeasePages/insurance/edit/InsuranceYupSchema.ts

```diff
@@ -4,15 +4,19 @@ import * as Yup from 'yup';
 export const InsuranceYupSchema = Yup.object().shape({
   insurances: Yup.array().of(
     Yup.object().shape({
-      coverageLimit: Yup.number().max(
-        MAX_SQL_MONEY_SIZE,
-        `Coverage Limit must be less than ${MAX_SQL_MONEY_SIZE}`,
+      coverageLimit: Yup.lazy(value =>
+        value === ''
+          ? Yup.string()
+          : Yup.number().max(
+              MAX_SQL_MONEY_SIZE,
+              `Coverage Limit must be less than ${MAX_SQL_MONEY_SIZE}`,
+            ),
```

Comment:
This support a field that is not required so it can be a number or it can be the empty string ""